src/pages/ArticlePage.jsx

```diff
@@ -1,39 +1,65 @@
 import React from "react";
 import { useParams } from "react-router-dom";
 import { useState, useEffect } from "react";
-import { fetchArticleById } from "../api";
+import { fetchArticleById, fetchCommentsByArticleId } from "../api";
 
 function ArticlePage() {
-  const { article_id } = useParams();
-  const [article, setArticle] = useState(null);
-  const [isLoading, setIsLoading] = useState(true);
+    const { article_id } = useParams();
+    const [article, setArticle] = useState(null);
+    const [comments, setComments] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+  
+    useEffect(() => {
+      fetchArticleById(article_id)
+        .then((fetchedArticle) => {
+          setArticle(fetchedArticle);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+  
+      fetchCommentsByArticleId(article_id)
+        .then((fetchedComments) => {
+          setComments(fetchedComments);
+          setIsLoading(false); 
+        })
+        .catch((err) => {
+          console.error(err); 
+          setIsLoading(false);
+        });
+    }, [article_id]); 
```

Comment:
It maybe worth splitting the API call for comments up into a separate component and make a separate API call. This way, you don't have to wait for everything to be retrieved until the article loads, you could make a call for the article, render it _then_ make a call to the comments, then render those.